src/main/cpp/apple_fsnotifier.cpp

```diff
@@ -78,13 +84,14 @@ static void callback(ConstFSEventStreamRef streamRef,
             type = FILE_EVENT_REMOVED;
         } else if (IS_SET(flags, kFSEventStreamEventFlagItemCreated)) {
             type = FILE_EVENT_CREATED;
-        } else if (IS_SET(flags, kFSEventStreamEventFlagItemInodeMetaMod)) {
-            // File locked
+        } else if (IS_ANY_SET(flags,
+                kFSEventStreamEventFlagItemInodeMetaMod // file locked
+                | kFSEventStreamEventFlagItemFinderInfoMod
+                | kFSEventStreamEventFlagItemChangeOwner
+                | kFSEventStreamEventFlagItemXattrMod)) {
             type = FILE_EVENT_MODIFIED;
```

Comment:
Shall we say `METADATA_MODIFIED` instead?

Reply:
Yeah, I guess we could do that, though that's changing the API a bit.

I wonder how useful the distinction for us is in any case. Would you consider a locking event to be a metadata change or content change (assuming the file is now unavailable to be read by Gradle)? Same for owner change.

Reply:
I mean, Gradle would still need to invalidate its VFS cache on `METADATA_CHANGED`, right? We can distinguish even so.

Reply:
We can leave it as `MODIFIED` for now. I would return everything which doesn't change the contents as `METADATA_CHANGED`. Note that the `kFSEventStreamEventFlagItemModified` we capture above also may have some metadata change flag set.